Refactor/Main.lean

```diff
@@ -0,0 +1,134 @@
+/-
+Copyright (c) 2024 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+-/
+import Lean.Util.Paths
+import Lean.Elab.Frontend
+
+/-! # `lake exe refactor` command
+
+This command will run a refactoring task over mathlib (or the specified modules) and all their
+dependencies. The task itself must be edited into this file (see `runRefactoring` below).
+
+-/
+
+def help : String := "Mathlib4 refactoring tool
+Usage: lake exe refactor <MODULE>..
+
+Arguments:
+  <MODULE>
+    A module path like `Mathlib`. All files transitively reachable from the
+    provided module(s) will be checked.
+
+Note that this tool will not work on its own. Edit `runRefactoring` in
+`Refactor/Main.lean` to add the refactoring to apply before running this command.
+"
+
+open Lean Elab Command Frontend
+
+instance : ToJson String.Pos := ⟨fun p => toJson p.1⟩
+instance : FromJson String.Pos := ⟨fun p => String.Pos.mk <$> fromJson? p⟩
+deriving instance ToJson for String.Range
+deriving instance FromJson for String.Range
+
+structure Edit where
+  replacement : String
+  range : String.Range
+  deriving ToJson, FromJson
+
+def runRefactoring (_cmd : TSyntax `command) : CommandElabM (List Edit) := do
+  -- INSERT LEAN CODE HERE
+  panic! "no refactoring active, edit Refactor/Main.lean to add one"
+  -- let `(command| #align $_ $_) := cmd | return []
+  -- let some range := cmd.raw.getRange? | return []
+  -- pure [⟨"", range⟩]
```

Comment:
Could you add an explanation to the module docstring how to exactly use this tool? The comments basically demonstrate it, but I think it would be useful for less experienced users to have these instructions in text form, too.